service/storage_service.cc

```diff
@@ -5740,7 +5740,7 @@ future<> storage_service::cleanup_tablet(locator::global_tablet_id tablet) {
                 if (!trinfo->pending_replica) {
                     throw std::runtime_error(fmt::format("Tablet {} has no pending replica", tablet));
                 }
-                if (trinfo->pending_replica->host != tm->get_my_id()) {
+                if (trinfo->pending_replica && trinfo->pending_replica->host != tm->get_my_id()) {
```

Comment:
The check for tringo->pending_replica had been done already

Reply:
reverted.